app/forms/bulk_submission_form.rb

```diff
@@ -78,4 +92,95 @@ def file_size(file)
     return 0 if file.nil?
     File.size(file&.tempfile)
   end
+
+  def csv
+    @csv ||= CSV.parse(File.read(uploaded_file), headers: true)
+  rescue StandardError
+    nil
+  end
+
+  def file_content
+    return unless csv && file_length
```

Comment:
row_count might be a better name but i like file_length too so 🤷

Reply:
have amended file_length to row_count